src/NUnitFramework/framework/Internal/Builders/NUnitTestCaseBuilder.cs

```diff
@@ -224,20 +250,28 @@ private static bool CheckTestMethodSignature(TestMethod testMethod, MethodInfoCa
                 if (typeArguments is null && (
                     arglist is null || !new GenericMethodHelper(testMethod.Method.MethodInfo).TryGetTypeArguments(arglist, out typeArguments)))
                 {
-                    return MarkAsNotRunnable(testMethod, "Unable to determine type arguments for method");
+                    MarkAsNotRunnable(testMethod, "Unable to determine type arguments for method");
+                    return parms;
                 }
 
                 testMethod.Method = testMethod.Method.MakeGenericMethod(typeArguments);
                 parameters = testMethod.Method.GetParameters();
             }
 
             if (parms is not null && parms.TestName is not null && string.IsNullOrWhiteSpace(parms.TestName))
-                return MarkAsNotRunnable(testMethod, "Test name cannot be all white-space or empty.");
+            {
+                MarkAsNotRunnable(testMethod, "Test name cannot be all white-space or empty.");
+                return parms;
+            }
 
             if (arglist is not null && parameters is not null)
-                TypeHelper.ConvertArgumentList(arglist, parameters);
+            {
+                // Convert the arguments for this specific test method
+                // Since we cloned parms earlier, this only affects this test method's copy
+                parms!.Arguments = TypeHelper.ConvertArgumentList(arglist, parameters);
+            }
 
-            return true;
+            return parms;
```

Comment:
The method always returns the parameter _parms_. It might as well return nothing.
Previously the (boolean) return type was ignored.

Reply:
Done, except postpone, will look into that